src/main/java/org/springframework/data/redis/connection/jedis/JedisStreamCommands.java

```diff
@@ -0,0 +1,496 @@
+
+	private final JedisConnection connection;
+
+	JedisStreamCommands(JedisConnection connection) {
+		this.connection = connection;
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.redis.connection.RedisStreamCommands#xAck(byte[], String, org.springframework.data.redis.connection.stream.RecordId[])
+	 */
+	@Override
+	public Long xAck(byte[] key, String group, RecordId... recordIds) {
+		Assert.notNull(key, "Key must not be null!");
+		Assert.hasText(group, "Group name must not be null or empty!");
+		Assert.notNull(recordIds, "recordIds must not be null!");
+
+		return connection.invoke().just(BinaryJedis::xack, MultiKeyPipelineBase::xack, key, JedisConverters.toBytes(group),
+				entryIdsToBytes(recordIds));
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.redis.connection.RedisStreamCommands#xAdd(MapRecord, XAddOptions)
+	 */
+	@Override
+	public RecordId xAdd(MapRecord<byte[], byte[], byte[]> record, XAddOptions options) {
+		Assert.notNull(record, "Record must not be null!");
+		Assert.notNull(record.getStream(), "Stream must not be null!");
+
+		byte[] id = JedisConverters.toBytes(record.getId().getValue());
+		Long maxLength = Long.MAX_VALUE;
+		if (options.hasMaxlen()) {
+			maxLength = options.getMaxlen();
+		}
+
+		return connection.invoke().from(BinaryJedis::xadd, MultiKeyPipelineBase::xadd, record.getStream(), id,
+				record.getValue(), maxLength, false).get(it -> RecordId.of(JedisConverters.toString(it)));
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.redis.connection.RedisStreamCommands#xClaimJustId(byte[], java.lang.String, java.lang.String, org.springframework.data.redis.connection.RedisStreamCommands.XClaimOptions)
+	 */
+	@Override
+	public List<RecordId> xClaimJustId(byte[] key, String group, String newOwner, XClaimOptions options) {
+		throw new UnsupportedOperationException("Jedis does not support xClaimJustId.");
+	}
+
+	/*
+	 * (non-Javadoc)
+	 * @see org.springframework.data.redis.connection.RedisStreamCommands#xClaim(byte[], java.lang.String, java.lang.String, org.springframework.data.redis.connection.RedisStreamCommands.XClaimOptions)
+	 */
+	@Override
+	public List<ByteRecord> xClaim(byte[] key, String group, String newOwner, XClaimOptions options) {
+		Assert.notNull(key, "Key must not be null!");
+		Assert.notNull(group, "Group must not be null!");
+		Assert.notNull(newOwner, "NewOwner must not be null!");
+
+		long minIdleTime = -1L;
+		if (options.getMinIdleTime() != null) {
+			minIdleTime = options.getMinIdleTime().toMillis();
+		}
+		int retryCount = -1;
+		if (options.getRetryCount() != null) {
+			retryCount = options.getRetryCount().intValue();
+		}
+		long unixTime = -1L;
+		if (options.getUnixTime() != null) {
+			unixTime = options.getUnixTime().toEpochMilli();
+		}
+		try {
+			if (isPipelined()) {
+				pipeline(connection.newJedisResult(connection.getRequiredPipeline().xclaim(key, JedisConverters.toBytes(group),
```

Comment:
Any reason to not use `JedisInvoker` here?

Reply:
This is a real headache because the parameter types of the `BinaryJedis::xclaim` `MultiKeyPipelineBase::xclaim` two methods are not the same.